### DISCUSSION TITLE
Hosted online demo

### DISCUSSION THREAD

**Discussion by brenocq** [brenocq/implot3d#25] [General]
It would be great to have a hosted online demo for ImPlot3D where people can easily try it out. This would allow potential users to experience the capabilities of ImPlot3D without needing to set up the environment locally.

Ideally, this demo would be automatically updated with the `main` branch or in a release-based.

It would be nice for it to look like [this hosted demo of ImPlot](https://traineq.org/implot_demo/src/implot_demo.html) and this [ImGui Manual](https://pthom.github.io/imgui_manual_online/manual/imgui_manual.html) (both by @pthom), but it could also contain some extra demos from other repositories with more complex uses of ImPlot3D.

**Comment by pthom** [comment]
Hello and congrats for your project! 
I'm the author of the demo you mention (as well as [ImGui Manual](https://pthom.github.io/imgui_manual_online/manual/imgui_manual.html)) can help in setting this up if you need. 
Also I'll study the possibility to integrate your lib into Dear ImGui Bundle.

This will be in january, I guess.
Cheers

**Reply by brenocq** [reply]
Hi @pthom! Thank you a lot for offering your help, your demos are just amazing!

Right now I can't focus on the online demo because many problems were reported in #24, so I'll be focusing on them. If you start working on the online demo please let us know here! If you need my help or have any questions I'll try to reply in this discussion as fast as I can :)

It would be great to see ImPlot3D in the Dear ImGui Bundle! Please let me know if you have any problems while integrating it. It should be very similar to how you integrated ImPlot.

**Reply by pthom** [reply]
Hi @brenocq 
I did a first version where the demos from implot and implot3d are shown in the same app: the original implot demo, now also hosts a demo for implot3d.

See result here: https://traineq.org/implot_demo/src/implot_demo.html

Happy new year!

**Reply by brenocq** [reply]
Hey @pthom, happy new year!

I just tested it and it looks great! It's cool that it even worked on my phone haha. Thanks for taking the time to work on it :)

Feel free to create a PR adding your link to the README when you think it is ready. It would be nice to make the **Demos** text (red rectangle in the image below) link to your online demo. We could also cite your demo in the `## 🎨 Demos` section.

![image](https://github.com/user-attachments/assets/09472840-a9e8-4966-a4c5-9ee62776001a)

**Comment by brenocq** [comment]
Thanks for the amazing work @pthom! Since your PR #43 was just merged into `dev` I'm closing this discussion.

**Comment by pthom** [comment]
For info, the demo now provides a button to quickly access [implot3d_demo.cpp](https://github.com/brenocq/implot3d/blob/main/implot3d_demo.cpp) code on Github. It also provides a button to access the related [implot3d_demo.py](https://github.com/pthom/imgui_bundle/blob/main/bindings/imgui_bundle/demos_python/demos_implot3d/implot3d_demo.py) file (available if using Dear ImGui Bundle python bindings).

See screenshot below:

<img width="549" alt="image" src="https://github.com/user-attachments/assets/2844e1ac-3dfd-4ac1-bd42-44bdd50453ca" />
